### PR TITLE
Rename `Material` component to `AlbedoFactor`

### PR DESCRIPTION
**Pull request by emilk**
### What
* Closes https://github.com/rerun-io/rerun/issues/6829

⚠️ This is a breaking change!

This renames the `Material` component to `AlbedoFactor`, which now simply wraps a `datatypes.Rgba32`.

Consequently, the field `mesh_material` in `Mesh3D` is now named `albedo_factor`

When constructing a `Mesh3D`:
* C++ & Rust: `.with_mesh_material(Material::from_albedo_factor(color))` -> `with_albedo_factor(color)`
* Python: `mesh_material=rr.Material(albedo_factor=color)` -> `albedo_factor=color`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6841?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6841?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!
* [x] I have run the `raw_mesh` examples
* [x] I have asked for a final `@rerun-bot full-check` and checked the results

- [PR Build Summary](https://build.rerun.io/pr/6841)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

**Comment by emilk** [comment]
@rerun-bot full-check

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## [Unreleased](https://github.com/rerun-io/rerun/compare/latest...HEAD)
 
+# ⚠️ Breaking changes
+* `mesh_material: Material` has been renamed to `albedo_factro: AlbedoFactor` [#6841](https://github.com/rerun-io/rerun/pull/6841)
```
Comment (Wumpf) [inline CHANGELOG.md:6]: good call to add it right away! Wondering if we should also add the same note in the migration guide from the start? I know its redundant, but we already have it and it's a good place to dump the micro code snippet you already put on the PR description :)

File: crates/viewer/re_edit_ui/src/albedo_factor.rs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    let rgba = value.as_ref().0;
+
+    if let Some(value) = value.as_mut() {
+        let mut edit_color = Color(rgba);
+        let response = edit_color_ui(ctx, ui, &mut MaybeMutRef::MutRef(&mut edit_color));
+        value.0 = edit_color.0;
+        response
+    } else {
+        edit_color_ui(ctx, ui, &mut MaybeMutRef::Ref(&Color(rgba)))
+    }
```
Comment (Wumpf) [inline crates/viewer/re_edit_ui/src/albedo_factor.rs:20]: nit: This is fine as is, but we'll have this again, so we might as well add a `edit_rgba32`
```
pub fn edit_rgba32(
    _ctx: &re_viewer_context::ViewerContext<'_>,
    ui: &mut egui::Ui,
    value: &mut MaybeMutRef<'_, impl std::ops::DerefMut<Target = datatypes::Rgba32>>,
) -> egui::Response {
```
which forwards to non-generic `edit_rgba32_impl`.
The registration call in lib.rs then becomes `registry.add_singleline_edit_or_view::<AlbedoFactor>(edit_rgba32);`

I.e. the same pattern as with a bunch of the other more generic UIs

**Review by Wumpf — APPROVED** [review]
👍

**Comment by emilk** [comment]
@rerun-bot full-check

**Pull request merged**